### PR TITLE
SDIT-1176 NomisApiV1 and OffenderFinePayment

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/NomisApiV1Resource.java [hunk 1]
```diff
@@ -199,7 +199,7 @@ public Events getOffenderEvents(@Size(max = 3) @RequestParam("prison_id") @Param
             "<p>If the account was previously closed then it will be closed again.</p>" +
             "<p>If the offender has been released then the funds are transferred to NACRO. Based on the Nomis Clear Inactive accounts screen (OTDCLINA).</p>")
     @PostMapping("/prison/{previous_prison_id}/offenders/{noms_id}/transfer_transactions")
-    @HasWriteScope
+    @PreAuthorize("hasAnyRole('SYSTEM_USER','NOMIS_API_V1', 'UNILINK') and hasAuthority('SCOPE_write')")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/NomisApiV1Resource.java:202]: Have to override the whole preAuthorize

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 2]
```diff
@@ -124,7 +159,6 @@ public class BookingService {
     private final VisitVisitorRepository visitVisitorRepository;
     private final SentenceRepository sentenceRepository;
     private final SentenceTermRepository sentenceTermRepository;
-    private final SentenceCalcTypeRepository sentenceCalcTypeRepository;
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:127]: This was not used

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 3]
```diff
@@ -287,7 +319,7 @@ public Map<Long, List<String>> getBookingAlertSummary(final List<Long> bookingId
     public uk.gov.justice.hmpps.prison.api.support.Page<ScheduledEvent> getBookingActivities(final Long bookingId, final LocalDate fromDate, final LocalDate toDate, final long offset, final long limit, final String orderByFields, final Order order) {
         validateScheduledEventsRequest(fromDate, toDate);
 
-        final var sortFields = StringUtils.defaultString(orderByFields, "startTime");
+        final var sortFields = Objects.toString(orderByFields, "startTime");
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:322]: Intellij deprecated warnings

File: src/main/java/uk/gov/justice/hmpps/prison/service/v1/NomisApiV1Service.java [hunk 4]
```diff
@@ -320,7 +318,7 @@ public List<AccountTransaction> getAccountTransactions(final String prisonId, fi
         return financeV1Repository.getAccountTransactions(prisonId, nomsId, accountType, fromDate, toDate)
                 .stream()
                 .map(t -> AccountTransaction.builder()
-                        .id("" + t.getTxnId() + "-" + t.getTxnEntrySeq())
+                        .id(t.getTxnId() + "-" + t.getTxnEntrySeq())
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/v1/NomisApiV1Service.java:321]: Intellij warning: redundant ""

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/v1/impl/NomisApiV1ResourceIntTest.java [hunk 5]
```diff
@@ -218,84 +236,85 @@ public void createTransaction() {
...
+                            "iep_level": {
+                                "code": "STD",
+                                "desc": "Standard"
+                            }
+                        },
+                        "case_details": {
+                            "personal_officer": "Griffine, Ymmnatpher"
+                        }
+                    }
+                 }""";
```
Comment (steverendell) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/v1/impl/NomisApiV1ResourceIntTest.java:317]: Intellij suggestion

File: src/test/java/uk/gov/justice/hmpps/prison/util/JwtAuthenticationHelper.kt [hunk 6]
```diff
@@ -60,12 +53,11 @@ class JwtAuthenticationHelper {
...
-      .addClaims(claims)
-      .setExpiration(Date(System.currentTimeMillis() + expiryTime.toMillis()))
-      .signWith(keyPair.private, SignatureAlgorithm.RS256)
-      .setHeaderParam("typ", "JWT")
-      .setHeaderParam("kid", "dps-client-key")
+      .id(UUID.randomUUID().toString())
+      .claims(claims)
+      .expiration(Date(System.currentTimeMillis() + expiryTime.toMillis()))
+      .signWith(keyPair.private, Jwts.SIG.RS256)
+      .header().add("typ", "JWT").add("kid", "dps-client-key").and()
```
Comment (steverendell) [inline src/test/java/uk/gov/justice/hmpps/prison/util/JwtAuthenticationHelper.kt:60]: Intellij deprecated warnings

**Review by petergphillips — APPROVED** [review]
LGTM

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**